System.Containers/ContainerHelpers.cs

```diff
@@ -0,0 +1,97 @@
+namespace System.Containers
+{
+    public static class ContainerHelpers
+    {
+        private static string[] knownImageSeparators = { "-", ".", "_" };
+
+        /// <summary>
+        /// Given some "fully qualified" image name (e.g. mcr.microsoft.com/dotnet/runtime), return
+        /// a valid UriBuilder. This means appending 'https' if the URI is not absolute, otherwise UriBuilder will throw.
+        /// </summary>
+        /// <param name="containerBase"></param>
+        /// <returns>A UriBuilder with the given containerBase, or, if containerBase is relative, https:// + containerBase</returns>
+        private static UriBuilder? ContainerImageToUriBuilder(string containerBase)
```

Comment:
Does this need to return the builder or could it return `Builder.Uri`? Looks like that's all that's used below.